pyramid_oereb/contrib/print_proxy/mapfish_print.py

```diff
@@ -487,3 +494,82 @@ def _multilingual_text(self, parent, name):
         if name in parent:
             lang_obj = dict([(e['Language'], e['Text']) for e in parent[name]])
             parent[name] = lang_obj[self._language]
+
+    def _sort_sub_themes(self, restrictions):
+        # split restrictions by theme codes
+        split_by_theme_code = self._split_restrictions_by_theme_code(restrictions)
+
+        # sort sub themes of the same theme
+        for theme_code in split_by_theme_code:
+            sub_themes = []
+            non_sub_themes = []
+            for restriction in split_by_theme_code[theme_code]:
+                if restriction.get('Split_SubTheme', False):
+                    sub_themes.append(restriction)
+                else:
+                    non_sub_themes.append(restriction)
+            # only sort if there are multiple sub themes
+            if len(sub_themes) > 1:
+                sorter, params = self._get_sorter(theme_code)
+                sub_themes = sorter.sort(sub_themes, params)
+            split_by_theme_code[theme_code] = non_sub_themes + sub_themes
+
+        # sort + flatten the splitted themes again
+        sorted_restrictions = []
+        for theme in Config.get_themes():
+            if theme.code in split_by_theme_code:
+                sorted_restrictions += split_by_theme_code[theme.code]
+
+        return sorted_restrictions
+
+    @staticmethod
+    def _split_restrictions_by_theme_code(restrictions):
+        """
+        Args:
+            restrictions (list): array of restrictions
+
+        Returns:
+            (dict) restrictions split up by theme code
+        :return:
+        """
+        split_by_theme_code = {}
+        for restriction in restrictions:
+            theme_code = restriction.get('Theme_Code')
+            if theme_code in split_by_theme_code:
+                split_by_theme_code[theme_code].append(restriction)
+            else:
+                split_by_theme_code[theme_code] = [restriction]
+        return split_by_theme_code
+
+    @staticmethod
+    def _load_sorter(module, class_name):
```

Comment:
This file was already full of private static methods.

I don't see any need to make them public, because they are internal helpers that are not needed outside of this class. You could argue about the static part, but they don't use any object properties. We could refactor it into a separate file, but then I would rather start with the gigantic main function (convert_to_printable_extract)